src/JukeBox.cpp

```diff
@@ -45,34 +65,158 @@ TResult<vector<BaseTrack>> JukeBox::queryTracks(string const &searchPattern,
 }
 
 TResult<QueueStatus> JukeBox::getCurrentQueues(TSessionID const &) {
-  return Error(ErrorCode::NotImplemented,
-               "getCurrentQueues is not implemented yet");
+  QueueStatus qs;
+  //  qs.adminQueue = mDataStore.getQueue(QueueType::Admin);
+  //  if (holds_alternative<Error>(qs.adminQueue))
+  //    return get<Error>(qs.adminQueue);
+  //
+  //  qs.normalQueue = mDataStore.getQueue(QueueType::Normal);
+  //  if (holds_alternative<Error>(qs.normalQueue))
+  //    return get<Error>(qs.normalQueue);
+  //
+  //  qs.currentTrack = mDataStore.getPlayingTrack();
+  //  if (holds_alternative<Error>(qs.currentTrack))
+  //    return get<Error>(qs.currentTrack);
+  //
+  return qs;
 }
 
 TResultOpt JukeBox::addTrackToQueue(TSessionID const &sid,
                                     TTrackID const &trkid,
                                     QueueType type) {
+  /* TODO:
+   * where does this function get the actual Track object/information from?
```

Comment:
using the track ID the `MusicBackend` should deliver you the full `BaseTrack` object